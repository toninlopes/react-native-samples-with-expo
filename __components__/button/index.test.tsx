import React from 'react';
import renderer from 'react-test-renderer';

import { Button } from '.';

describe('Button', () => {
  test('When there is a title.', () => {
    const button = renderer.create((
      <Button>
        <Button.Title>{'Title'}</Button.Title>
      </Button>
    )).toJSON();
    expect(button).toMatchSnapshot();
  });

  test('When there are title and icon.', () => {
    const button = renderer.create((
      <Button>
        <Button.Title>{'Title'}</Button.Title>
        <Button.Icon name='timer-outline' />
      </Button>
    )).toJSON();
    expect(button).toMatchSnapshot();
  });
});