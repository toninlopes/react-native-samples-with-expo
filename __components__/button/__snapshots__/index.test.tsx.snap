// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button When there are title and icon. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "silver",
      "borderRadius": 8,
      "flexDirection": "row",
      "height": 42,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 6,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "fontSize": 18,
      }
    }
  >
    Title
  </Text>
  <View
    style={
      {
        "paddingHorizontal": 12,
      }
    }
  >
    <Text />
  </View>
</View>
`;

exports[`Button When there is a title. 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "silver",
      "borderRadius": 8,
      "flexDirection": "row",
      "height": 42,
      "justifyContent": "center",
      "opacity": 1,
      "padding": 6,
    }
  }
>
  <Text
    style={
      {
        "alignSelf": "center",
        "fontSize": 18,
      }
    }
  >
    Title
  </Text>
</View>
`;
